### ISSUE TITLE
`|||` breaks instance declarations

### ISSUE THREAD

**Issue by nicolabotta** [idris-lang/Idris-dev#1943]
The code
````
> ||| The singletons
> class Lala a where
>   lala : (x : a) -> (y : a) -> x = y

> -- ||| () is a singleton
> instance Lala () where
>   lala () () = Refl
```
type checks but uncommenting the line `> -- ||| () is a singleton` yields the error
````
- + Errors (1)
 `-- ./instance.lidr line 6 col 0:
      error: expected: "class",
         "|||", accessibility modifier
     instance Lala () where 
     ^
```

**Comment by david-christiansen** [comment]
This is because individual instances can't have documentation strings. I considered it to not be a good idea, because there's not really a syntax to look up those docs, but on further reflection I think that this assumption was bad.

Perhaps a sensible interface might be something like:
```
Idris> :doc Uninhabited
Type class Uninhabited
    A canonical proof that some type is empty

Parameters:
    t

Methods:
    uninhabited : Uninhabited t => t -> Void
        If I have a t, I've had a contradiction
       
Instances:
    Uninhabited Void:
      Void is the canonical empty type, thus it is trivially uninhabited.
    Uninhabited (0 = S n)
      The constructors of Nat are disjoint.
```

where the instance docstrings would be shown in the documentation for the class itself. This would also be nice for named instances, come to think of it.

Thanks for a good idea!

Marking this as "low-hanging fruit" because it should be a matter of following the code for the existing docstring handling and adding in a bit more.

**Comment by nicolabotta** [comment]
Thanks a lot David! Best

**Comment by david-christiansen** [comment]
This was fixed by #1983. Thanks @adamse!
